src/index.ts

```diff
@@ -99,7 +101,7 @@ export const getSession: GetSession = (...args) => getInstance().getSession(...a
 export const getAccessToken: GetAccessToken = (...args) => getInstance().getAccessToken(...args);
 export const withApiAuthRequired: WithApiAuthRequired = (...args) => getInstance().withApiAuthRequired(...args);
 export const withPageAuthRequired: WithPageAuthRequired = (...args: any[]): any =>
-  withPageAuthRequiredFactory(getSession)(...args);
+  withPageAuthRequiredFactory(config, getSession)(...args);
```

Comment:
The empty config you're passing here won't be populated by the time it's used because you're [querying the config](https://github.com/auth0/nextjs-auth0/blob/5a909dac5a3bc0a94120a5220c59322564452078/src/helpers/with-page-auth-required.ts#L103) before `getSession` (which calls `initAuth0`) [is called](https://github.com/auth0/nextjs-auth0/blob/5a909dac5a3bc0a94120a5220c59322564452078/src/helpers/with-page-auth-required.ts#L106)

You also can't reassign the config, you'd need to mutate it `Object.assign(config, nextConfig)` for it to work.

Either way, I think it's too brittle to initialise an empty config then expect it to be populated when you want to use it.

For this special case, the easiest thing to do is probably just pass the `loginUrl`

```js
export const withPageAuthRequired: WithPageAuthRequired = (...args: any[]): any =>
  withPageAuthRequiredFactory({ loginUrl: process.env. NEXT_PUBLIC_AUTH0_LOGIN || '/api/auth/login'}, getSession)(...args);
```

You can DRY this up by creating a `getLoginUrlFromEnv` type function in `config.ts` if you don't want to do the `process.env. NEXT_PUBLIC_AUTH0_LOGIN || '/api/auth/login'` twice

Or if you want the signature for `withPageAuthRequiredFactory` to still use the whole `NextConfig` object, you could create a `getNextConfigFromEnv` type function in `config.ts`.

Let me know if you want me to go through this with you

Reply:
Thanks changed in https://github.com/auth0/nextjs-auth0/pull/268/commits/2b4b4c6e8d9dc23051e22ba777b22aa9a7bc8cda.